sw/device/lib/testing/i2c_testutils.c

```diff
@@ -317,3 +314,11 @@ status_t i2c_testutils_wait_host_idle(const dif_i2c_t *i2c) {
   } while (!status.host_idle);
   return OK_STATUS();
 }
+
+status_t i2c_testutils_wait_transaction_finish(const dif_i2c_t *i2c) {
+  dif_i2c_status_t status;
+  do {
+    TRY(dif_i2c_get_status(i2c, &status));
+  } while (!status.fmt_fifo_empty);
```

Comment:
Small tweak (but maybe could be made in another PR):

Can we make use of the FMT_FIFO threshold and interrupt instead? This would represent the more typical pathway in practice, I think. We would still poll here, but it'd be the interrupt bit instead.